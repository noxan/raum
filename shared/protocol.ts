export enum Action {
  INSERT = 1,
  UPDATE = 2,
  FIND = 3,
  DELETE = 4,
}

export const encodeMessage = (action: Action, data?: object) => {
  return `${action}@${JSON.stringify(data)}`;
};
