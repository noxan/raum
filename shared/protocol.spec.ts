import { Action, encodeMessage } from './protocol';

test('encodeMessage', () => {
  const message = encodeMessage(Action.INSERT);
  expect(message).toMatchSnapshot();
});
