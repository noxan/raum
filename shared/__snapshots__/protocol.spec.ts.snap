// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`encodeMessage 1`] = `"1"`;
